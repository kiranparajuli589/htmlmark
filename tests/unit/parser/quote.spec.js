import { MDP } from "../../../lib/index.js"


describe("Quote Parsing", () => {
	it("should parse multiline quote with the same depth and indent", () => {
		const lines = [
			"> > > zero f",
			"> > one f",
			">>>",
			"> one",
			"> # two",
			"> > three",
			"> > > four",
			"> > > > d-five",
			"> > # five",
			"> > ## six",
			"> > > > seven",
			"simple para"
		]
		const html = MDP.h(lines)
		expect(html).toMatchSnapshot()
	})
	it("should detect the circular quote", () => {
		const lines = [
			"> one",
			"> > two",
			"> > > three",
			"> > > > four",
			"> > > > > five",
			"> >>>",
			"> >>> four",
			"> >>",
			"> >> three",
			"> >",
			"> > two",
			">",
			"> one"
		]
		const html = MDP.h(lines)
		expect(html).toMatchSnapshot()
	})
	it.each([
		"> quote **one** with *two*",
		"  > quote `four` with ~~five~~",
		"> quote [link-title](link-url) with *two*"
	])("should deep tokenize quote with zero depth", (line) => {
		const html = MDP.h([line])
		expect(html).toMatchSnapshot()
	})
	it.each([
		"> > quote **one** with *two*",
		"  > > > quote `four` with ~~five~~",
		"> > > > quote [link-title](link-url) with *two*"
	])("should deep tokenize quote with multiple depth", (line) => {
		const html = MDP.h([line])
		expect(html).toMatchSnapshot()
	})
	it("should parse the quote with list inside", () => {
		const lines = [
			"> - one",
			"> - two"
		]
		const html = MDP.h(lines)
		expect(html).toMatchSnapshot()
	})
	it("should find the list and codeblock inside", () => {
		const lines = [
			"> ## This is a header.",
			">",
			"> 1.   This is the first list item.",
			"> 2.   This is the second list item.",
			">",
			"> Here's some example code:",
			">",
			">    return shell_exec(\"echo $input | $markdown_script\");"
		]
		const html = MDP.h(lines)
		expect(html).toMatchSnapshot()
	})
})
