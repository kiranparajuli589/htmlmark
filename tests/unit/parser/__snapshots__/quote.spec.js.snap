// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Parsing should deep tokenize quote with multiple depth 1`] = `
"<blockquote>
<blockquote>
<p>quote <strong>one</strong> with <em>two</em></p>
</blockquote>
</blockquote>
"
`;

exports[`Quote Parsing should deep tokenize quote with multiple depth 2`] = `
"<blockquote>
<blockquote>
<blockquote>
<p>quote <code>four</code> with <s>five</s></p>
</blockquote>
</blockquote>
</blockquote>
"
`;

exports[`Quote Parsing should deep tokenize quote with multiple depth 3`] = `
"<blockquote>
<blockquote>
<blockquote>
<blockquote>
<p>quote <a href='link-url'>link-title</a> with <em>two</em></p>
</blockquote>
</blockquote>
</blockquote>
</blockquote>
"
`;

exports[`Quote Parsing should deep tokenize quote with zero depth 1`] = `
"<blockquote>
<p>quote <strong>one</strong> with <em>two</em></p>
</blockquote>
"
`;

exports[`Quote Parsing should deep tokenize quote with zero depth 2`] = `
"<blockquote>
<p>quote <code>four</code> with <s>five</s></p>
</blockquote>
"
`;

exports[`Quote Parsing should deep tokenize quote with zero depth 3`] = `
"<blockquote>
<p>quote <a href='link-url'>link-title</a> with <em>two</em></p>
</blockquote>
"
`;

exports[`Quote Parsing should detect the circular quote 1`] = `
"<blockquote>
<p>one</p><blockquote>
<p>two</p><blockquote>
<p>three</p><blockquote>
<p>four</p><blockquote>
<p>five</p>
</blockquote>
<br>
<p>four</p>
</blockquote>
<br>
<p>three</p>
</blockquote>
<br>
<p>two</p>
</blockquote>
<br>
<p>one</p>
</blockquote>
"
`;

exports[`Quote Parsing should find the list and codeblock inside 1`] = `
"<blockquote>
<h2>This is a header.</h2>
<br>
<ol>
<li><p>This is the first list item.</p></li>
<li><p>This is the second list item.</p></li>
</ol>
<br>
<p>Here's some example code:</p><br>
<pre><code>return shell_exec(&quot;echo $input | $markdown_script&quot;);</code></pre>

</blockquote>
"
`;

exports[`Quote Parsing should parse multiline quote with the same depth and indent 1`] = `
"<blockquote>
<blockquote>
<blockquote>
<p>zero f one f</p><br>

</blockquote>

</blockquote>
<p>one</p><h1>two</h1>
<blockquote>
<p>three</p><blockquote>
<p>four</p><blockquote>
<p>d-five</p>
</blockquote>

</blockquote>
<h1>five</h1>
<h2>six</h2>
<blockquote>
<blockquote>
<p>seven simple para</p>
</blockquote>
</blockquote>

</blockquote>

</blockquote>
"
`;

exports[`Quote Parsing should parse the quote with list inside 1`] = `
"<blockquote>
<ul>
<li><p>one</p></li>
<li><p>two</p></li>
</ul>

</blockquote>
"
`;
